### PR TITLE
Bridge asset locker

### PR DESCRIPTION
**Pull request by vovac12**

### CODE REVIEW

File: pallets/types/src/traits.rs [hunk 1]
```diff
@@ -322,3 +325,24 @@ impl<AssetId, Balance> BalancePrecisionConverter<AssetId, Balance, Balance> for
...
+pub trait BridgeAssetLocker<AccountId> {
+    type AssetId: Parameter + MaybeSerializeDeserialize;
+    type Balance: Parameter + AtLeast32BitUnsigned + MaybeSerializeDeserialize;
+
+    fn lock_asset(
+        network_id: GenericNetworkId,
+        asset_kind: AssetKind,
+        who: AccountId,
+        asset_id: Self::AssetId,
+        amount: Self::Balance,
```
Comment (Alexey-N-Chernyshov) [inline pallets/types/src/traits.rs]:
```suggestion
        asset_kind: &AssetKind,
        who: &AccountId,
        asset_id: &Self::AssetId,
        amount: &Self::Balance,
```
Is it better to pass by reference and not to clone on every call? As I understand, the method doesn't require ownership of the values. But I don't know best practice, if you know. could you provide me a link?

File: pallets/types/src/traits.rs [hunk 2]
```diff
@@ -322,3 +325,24 @@ impl<AssetId, Balance> BalancePrecisionConverter<AssetId, Balance, Balance> for
...
+    ) -> DispatchResult;
+
+    fn unlock_asset(
+        network_id: GenericNetworkId,
+        asset_kind: AssetKind,
+        who: AccountId,
+        asset_id: Self::AssetId,
+        amount: Self::Balance,
```
Comment (Alexey-N-Chernyshov) [inline pallets/types/src/traits.rs]: SAP

File: pallets/types/src/test_utils.rs [hunk 3]
```diff
@@ -147,3 +149,57 @@ impl BlockWithProofs {
             .collect()
     }
 }
+
+pub struct BridgeAssetLockerImpl<T>(PhantomData<T>);
```
Comment (Alexey-N-Chernyshov) [inline pallets/types/src/test_utils.rs:153]: It is used in mocks only, what about moving it to `pallets/substrate-app/src/mock.rs`?

**Review by N1ghtStorm — APPROVED** [review]

File: pallets/types/src/test_utils.rs (hunk 3 above)
Comment (vovac12) [reply pallets/types/src/test_utils.rs:153]: Also it's used in ERC20App, EthApp and MigrationApp

**Pull request merged**